Lesson 13/src/index.js

```diff
@@ -0,0 +1,131 @@
+// 1
+const nestedArray = [[1, 2], [3, [4, 5]], 6]
+const flatArray = nestedArray.flat(Infinity)
+
+console.log(flatArray)
+
+// 2
+
+const products = [
+  { id: 1, name: 'Телефон', inStock: true },
+  { id: 2, name: 'Ноутбук', inStock: false },
+  { id: 3, name: 'Часы', inStock: false }
+]
+
+function productInStock(productsArray) {
+  return productsArray.some((product) => product.inStock === true)
+}
+
+console.log(productInStock(products))
+
+// 3
+
+const transactions = [
+  { id: 1, type: 'income', amount: 500 },
+  { id: 2, type: 'expense', amount: 100 },
+  { id: 3, type: 'income', amount: 200 },
+  { id: 4, type: 'expense', amount: 50 }
+]
+
+function totalBalance(transactione) {
+  console.log(transactione.reduce((summary, currenTransaction) => {
+    if (currenTransaction.type === 'income') {
+      summary += currenTransaction.amount
+    } else summary -= currenTransaction.amount
+    return summary
+  }, 0))
+}
+
+totalBalance(transactions)
```

Comment:
```suggestion
const transactions = [
  { id: 1, type: 'income', amount: 500 },
  { id: 2, type: 'expense', amount: 100 },
  { id: 3, type: 'income', amount: 200 },
  { id: 4, type: 'expense', amount: 50 }
]

function calculateTotalBalance(transactionsArray) {
  console.log(transactionsArray.reduce((summary, currenTransaction) => {
    if (currenTransaction.type === 'income') {
      summary += currenTransaction.amount
    } else {
      summary -= currenTransaction.amount
    }

    return summary
  }, 0))
}

calculateTotalBalance(transactions)
```

заменил на название с глаголом, добавил фигурные скобки в else. функция верная - молодец 👍